src/http/web_socket/protocol.cr

```diff
@@ -58,15 +58,8 @@ class HTTP::WebSocket::Protocol
       (@buffer + @pos).copy_from(slice.to_unsafe, count)
       @pos += count
 
-      if @pos == @buffer.size
-        flush(final: false)
-      end
-
-      if count < slice.size
-        write(slice + count)
-      end
-
-      nil
+      flush(final: false) if @pos == @buffer.size
```

Comment:
I'd rather not have these changes. They're purely stylistic and there's not real benefit in either variant. Some prefer one, some prefer the other. Changing back and force for the sake of it, is just noise and avoidable.